### ISSUE TITLE
User unable to save user profile attributes

### ISSUE THREAD

**Issue by Malcolm-GetAHead** [keycloak/keycloak#21791]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

user-profile

### Describe the bug

Realm config
- accountv3 enabled
- User Profiles enabled
- Email as username on
- Edit username off

When a user attempts to save their profile (https://example.com/auth/realms/my-realm/account/#/) they receive a validation error on their username.

POST to `https://example.com/auth/realms/my-realm/account/`

```json
{ "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "username": "user@example.com", "firstName": "Name", "lastName": "Example", "emailVerified": false, "userProfileMetadata": { "attributes": [ { "name": "locale", "displayName": "locale", "required": false, "readOnly": false, "validators": {} }, { "name": "username", "displayName": "${username}", "required": true, "readOnly": true, "annotations": {}, "validators": { "username-prohibited-characters": { "ignore.empty.value": true }, "length": { "max": 255, "ignore.empty.value": true, "min": 3 } } }, { "name": "nickname", "displayName": "Nickname", "required": false, "readOnly": false, "annotations": {}, "validators": {} }, { "name": "firstName", "displayName": "${firstName}", "required": true, "readOnly": false, "annotations": {}, "validators": { "person-name-prohibited-characters": { "ignore.empty.value": true }, "length": { "max": 255, "ignore.empty.value": true } } }, { "name": "lastName", "displayName": "${lastName}", "required": true, "readOnly": false, "annotations": {}, "validators": { "person-name-prohibited-characters": { "ignore.empty.value": true }, "length": { "max": 255, "ignore.empty.value": true } } } ] }, "attributes": { "nickname": "AlteredField" } }
```

Response:
```json
{
    "errors": [
        {
            "field": "username",
            "errorMessage": "missingUsernameMessage",
            "params": [
                "${username}"
            ]
        },
        {
            "field": "email",
            "errorMessage": "error-user-attribute-required",
            "params": [
                "${email}"
            ]
        }
    ]
}
```

### Version

v22.0.0

### Expected behavior

User should be able to save their profile when

### Actual behavior

User receive error that their username is invalid/required when saving the profile

### How to Reproduce?

Setting the realm config as listed above

In Realm settings -> User Profile, for both username and email, if I set them as non-user-editable then I can save the settings but then during registration those fields are not present.

**Comment by pedroigor** [comment]
@Malcolm-GetAHead From what version are you migrating? Tested 21.1.2 and 19.0.3 and the behavior is the same.

Can you please also confirm users do not have e-mail set?

**Comment by Malcolm-GetAHead** [comment]
Hi @pedroigor , we started with v21 (forget sub version) but we are able to replicate with a fresh v22.0.1 install.

Users have a verified email address. 

We're currently mitigating the issue by enabling edit username.

**Comment by pedroigor** [comment]
@Malcolm-GetAHead Sent a PR that should fix this. See https://github.com/keycloak/keycloak/pull/22303.

**Comment by lalib** [comment]
Hey @pedroigor , when this is going to be released? I am having the same error for `22.0.1` version.
